crates/red_knot_server/src/server/api/requests/diagnostic.rs

```diff
@@ -66,11 +64,11 @@ fn to_lsp_diagnostic(message: &str) -> Diagnostic {
 
     let (range, message) = match words.as_slice() {
         [_filename, line, column, message] => {
-            let line = line.parse::<u32>().unwrap_or_default();
+            let line = line.parse::<u32>().unwrap_or_default().saturating_sub(1);
             let column = column.parse::<u32>().unwrap_or_default();
             (
                 Range::new(
-                    Position::new(line.saturating_sub(1), column.saturating_sub(1)),
+                    Position::new(line, column.saturating_sub(1)),
```

Comment:
This is just to make sure the diagnostics are highlighted for at most one character. Without this, it would spill over to the next line.